OMGVA PoS/Data layer/Repositories/Tax/TaxRepository.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OMGVA_PoS.Data_layer.Context;
+
+namespace OMGVA_PoS.Data_layer.Repositories.Tax;
+
+public class TaxRepository(OMGVADbContext context) : ITaxRepository
+{
+    public async Task<List<Models.Tax>> GetAllTaxesAsync()
+    {
+        return await context.Taxes
+            .ToListAsync();
+    }
+
+    public async Task<Models.Tax> GetTaxByIdAsync(long id)
+    {
+        return await context.Taxes.Where(tax => tax.Id == id).FirstOrDefaultAsync();
+    }
+
+    public async Task<Models.Tax> SaveTaxAsync(Models.Tax tax)
+    {
+        context.Taxes.Add(tax);
+        await context.SaveChangesAsync();
```

Comment:
For critical methods such as Save/Update/Delete I would wrap everything in a try-catch block (just to be sure).

Reply:
Will do